Ordered lists trigger change

### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

```
igor@Igor-Ajdiseks-MacBook-Pro ~ % terraform -v
Terraform v1.0.9
on darwin_amd64

# versions.tf
    cloudflare = {
      source  = "cloudflare/cloudflare"
      version = "~> 3.3.0"
    }
```

### Affected resource(s)

cloudflare_teams_list

### Terraform configuration files

```hcl
# initial resource
resource "cloudflare_teams_list" "test_list" {
 account_id  = var.account_id
  name        = "Test list"
  type        = "SERIAL"
  description = "Serial numbers for test"
  items       = [1,2,4,5]
}

# resource with item no. 3 added in ordered list
resource "cloudflare_teams_list" "test_list" {
 account_id  = var.account_id
  name        = "Test list"
  type        = "SERIAL"
  description = "Serial numbers for test"
  items       = [1,2,3,4,5]
}
```


### Debug output

```
% TF_LOG=debug terraform plan
2021-10-27T09:16:34.128+0200 [DEBUG] Adding temp file log sink: /var/folders/4q/8c4_y6ds3pzbwq80wp_r37kr0000gp/T/terraform-log2666747273
2021-10-27T09:16:34.128+0200 [INFO]  Terraform version: 1.0.9
2021-10-27T09:16:34.129+0200 [INFO]  Go runtime version: go1.17.2
2021-10-27T09:16:34.129+0200 [INFO]  CLI args: []string{"/usr/local/bin/terraform", "plan"}
2021-10-27T09:16:34.129+0200 [DEBUG] Attempting to open CLI config file: /Users/igor/.terraformrc
2021-10-27T09:16:34.129+0200 [INFO]  Loading CLI configuration from /Users/igor/.terraformrc
2021-10-27T09:16:34.129+0200 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2021-10-27T09:16:34.129+0200 [DEBUG] ignoring non-existing provider search directory /Users/igor/.terraform.d/plugins
2021-10-27T09:16:34.129+0200 [DEBUG] ignoring non-existing provider search directory /Users/igor/Library/Application Support/io.terraform/plugins
2021-10-27T09:16:34.129+0200 [DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
2021-10-27T09:16:34.129+0200 [INFO]  CLI command args: []string{"plan"}
2021-10-27T09:16:34.131+0200 [DEBUG] New state was assigned lineage "ff174555-224d-cf9a-65d6-c941b8953a2d"
2021-10-27T09:16:34.198+0200 [DEBUG] checking for provisioner in "."
2021-10-27T09:16:34.202+0200 [DEBUG] checking for provisioner in "/usr/local/bin"
2021-10-27T09:16:34.203+0200 [INFO]  Failed to read plugin lock file .terraform/plugins/darwin_amd64/lock.json: open .terraform/plugins/darwin_amd64/lock.json: no such file or directory
2021-10-27T09:16:34.203+0200 [INFO]  backend/local: starting Plan operation
2021-10-27T09:16:34.204+0200 [DEBUG] created provider logger: level=debug
2021-10-27T09:16:34.204+0200 [INFO]  provider: configuring client automatic mTLS
2021-10-27T09:16:34.214+0200 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0 args=[.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0]
2021-10-27T09:16:34.217+0200 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0 pid=19795
2021-10-27T09:16:34.217+0200 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0
2021-10-27T09:16:34.234+0200 [INFO]  provider.terraform-provider-cloudflare_v3.3.0: configuring server automatic mTLS: timestamp=2021-10-27T09:16:34.233+0200
2021-10-27T09:16:34.243+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.3.0: plugin address: network=unix address=/var/folders/4q/8c4_y6ds3pzbwq80wp_r37kr0000gp/T/plugin2241191398 timestamp=2021-10-27T09:16:34.243+0200
2021-10-27T09:16:34.243+0200 [DEBUG] provider: using plugin: version=5
2021-10-27T09:16:34.266+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-10-27T09:16:34.268+0200 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0 pid=19795
2021-10-27T09:16:34.268+0200 [DEBUG] provider: plugin exited
2021-10-27T09:16:34.268+0200 [INFO]  terraform: building graph: GraphTypeValidate
2021-10-27T09:16:34.268+0200 [DEBUG] ProviderTransformer: "cloudflare_teams_list.test_list" (*terraform.NodeValidatableResource) needs provider["registry.terraform.io/cloudflare/cloudflare"]
2021-10-27T09:16:34.268+0200 [DEBUG] ReferenceTransformer: "var.account_id" references: []
2021-10-27T09:16:34.268+0200 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/cloudflare/cloudflare\"]" references: []
2021-10-27T09:16:34.268+0200 [DEBUG] ReferenceTransformer: "cloudflare_teams_list.test_list" references: [var.account_id]
2021-10-27T09:16:34.269+0200 [DEBUG] Starting graph walk: walkValidate
2021-10-27T09:16:34.269+0200 [DEBUG] created provider logger: level=debug
2021-10-27T09:16:34.269+0200 [INFO]  provider: configuring client automatic mTLS
2021-10-27T09:16:34.278+0200 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0 args=[.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0]
2021-10-27T09:16:34.281+0200 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0 pid=19796
2021-10-27T09:16:34.281+0200 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0
2021-10-27T09:16:34.298+0200 [INFO]  provider.terraform-provider-cloudflare_v3.3.0: configuring server automatic mTLS: timestamp=2021-10-27T09:16:34.298+0200
2021-10-27T09:16:34.308+0200 [DEBUG] provider: using plugin: version=5
2021-10-27T09:16:34.308+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.3.0: plugin address: address=/var/folders/4q/8c4_y6ds3pzbwq80wp_r37kr0000gp/T/plugin191295859 network=unix timestamp=2021-10-27T09:16:34.308+0200
2021-10-27T09:16:34.331+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-10-27T09:16:34.332+0200 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0 pid=19796
2021-10-27T09:16:34.332+0200 [DEBUG] provider: plugin exited
2021-10-27T09:16:34.332+0200 [INFO]  backend/local: plan calling Plan
2021-10-27T09:16:34.332+0200 [INFO]  terraform: building graph: GraphTypePlan
2021-10-27T09:16:34.333+0200 [DEBUG] ProviderTransformer: "cloudflare_teams_list.test_list (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/cloudflare/cloudflare"]
2021-10-27T09:16:34.333+0200 [DEBUG] ReferenceTransformer: "cloudflare_teams_list.test_list (expand)" references: [var.account_id]
2021-10-27T09:16:34.333+0200 [DEBUG] ReferenceTransformer: "var.account_id" references: []
2021-10-27T09:16:34.333+0200 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/cloudflare/cloudflare\"]" references: []
2021-10-27T09:16:34.333+0200 [DEBUG] Starting graph walk: walkPlan
2021-10-27T09:16:34.333+0200 [DEBUG] created provider logger: level=debug
2021-10-27T09:16:34.333+0200 [INFO]  provider: configuring client automatic mTLS
2021-10-27T09:16:34.343+0200 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0 args=[.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0]
2021-10-27T09:16:34.345+0200 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0 pid=19797
2021-10-27T09:16:34.345+0200 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0
2021-10-27T09:16:34.362+0200 [INFO]  provider.terraform-provider-cloudflare_v3.3.0: configuring server automatic mTLS: timestamp=2021-10-27T09:16:34.362+0200
2021-10-27T09:16:34.371+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.3.0: plugin address: address=/var/folders/4q/8c4_y6ds3pzbwq80wp_r37kr0000gp/T/plugin3857694509 network=unix timestamp=2021-10-27T09:16:34.371+0200
2021-10-27T09:16:34.372+0200 [DEBUG] provider: using plugin: version=5
2021-10-27T09:16:34.399+0200 [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/cloudflare/cloudflare\"]" changed the config value, but that value is unused
2021-10-27T09:16:34.399+0200 [INFO]  provider.terraform-provider-cloudflare_v3.3.0: 2021/10/27 09:16:34 [INFO] Cloudflare Client configured for user: ***@****: timestamp=2021-10-27T09:16:34.399+0200
2021-10-27T09:16:34.399+0200 [INFO]  ReferenceTransformer: reference not found: "var.account_id"
2021-10-27T09:16:34.399+0200 [DEBUG] ReferenceTransformer: "cloudflare_teams_list.test_list" references: []
cloudflare_teams_list.test_list: Refreshing state... [id=002e2787-****-****-****-f88baba7a772]
2021-10-27T09:16:34.401+0200 [INFO]  provider.terraform-provider-cloudflare_v3.3.0: 2021/10/27 09:16:34 [DEBUG] Cloudflare API Request Details:
---[ REQUEST ]---------------------------------------
GET /client/v4/accounts/********************/gateway/lists/002e2787-****-****-****-f88baba7a772 HTTP/1.1
Host: api.cloudflare.com
User-Agent: terraform/1.0.9 terraform-plugin-sdk/2.8.0 terraform-provider-cloudflare/3.3.0
Content-Type: application/json
X-Auth-Email: ***@****
X-Auth-Key: **************
Accept-Encoding: gzip


-----------------------------------------------------: timestamp=2021-10-27T09:16:34.401+0200
2021-10-27T09:16:35.605+0200 [INFO]  provider.terraform-provider-cloudflare_v3.3.0: 2021/10/27 09:16:35 [DEBUG] Cloudflare API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Cf-Cache-Status: DYNAMIC
Cf-Ray: 6a4a23c4dd2c38bd-VIE
Content-Type: application/json; charset=UTF-8
Date: Wed, 27 Oct 2021 07:16:35 GMT
Expect-Ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
Server: cloudflare
Set-Cookie: __cflb=0H28vgHxwvgAQtjUGU56Rb8iNWZVUvXhuxQEQDTViSR; SameSite=Lax; path=/; expires=Wed, 27-Oct-21 09:46:36 GMT; HttpOnly
Set-Cookie: __cfruid=a07be4cf40743b924d9b51c3dac56290f948c6e5-1635318995; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
Vary: Accept-Encoding
X-Envoy-Upstream-Service-Time: 13

{
  "result": {
    "id": "002e2787-****-****-****-f88baba7a772",
    "name": "Test list",
    "description": "Serial numbers for test",
    "type": "SERIAL",
    "created_at": "2021-10-27T07:16:02Z",
    "updated_at": "2021-10-27T07:16:25Z",
    "count": 5
  },
  "success": true,
  "errors": [],
  "messages": []
}

-----------------------------------------------------: timestamp=2021-10-27T09:16:35.605+0200
2021-10-27T09:16:35.605+0200 [INFO]  provider.terraform-provider-cloudflare_v3.3.0: 2021/10/27 09:16:35 [DEBUG] Cloudflare API Request Details:
---[ REQUEST ]---------------------------------------
GET /client/v4/accounts/********************/gateway/lists/002e2787-****-****-****-f88baba7a772/items HTTP/1.1
Host: api.cloudflare.com
User-Agent: terraform/1.0.9 terraform-plugin-sdk/2.8.0 terraform-provider-cloudflare/3.3.0
Content-Type: application/json
X-Auth-Email: ***@****
X-Auth-Key: **************
Accept-Encoding: gzip


-----------------------------------------------------: timestamp=2021-10-27T09:16:35.605+0200
2021-10-27T09:16:36.630+0200 [INFO]  provider.terraform-provider-cloudflare_v3.3.0: 2021/10/27 09:16:36 [DEBUG] Cloudflare API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Cf-Cache-Status: DYNAMIC
Cf-Ray: 6a4a23cb4a9c38bd-VIE
Content-Type: application/json; charset=UTF-8
Date: Wed, 27 Oct 2021 07:16:36 GMT
Expect-Ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
Server: cloudflare
Set-Cookie: __cflb=0H28vgHxwvgAQtjUGU56Rb8iNWZVUvXhv86uR2GzKhw; SameSite=Lax; path=/; expires=Wed, 27-Oct-21 09:46:37 GMT; HttpOnly
Set-Cookie: __cfruid=8caad9254683d9547eafeca8baabb97f08a00737-1635318996; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
Vary: Accept-Encoding
X-Envoy-Upstream-Service-Time: 8

{
  "result": [
    {
      "value": "3",
      "created_at": "2021-10-27T07:16:25Z"
    },
    {
      "value": "5",
      "created_at": "2021-10-27T07:16:02Z"
    },
    {
      "value": "4",
      "created_at": "2021-10-27T07:16:02Z"
    },
    {
      "value": "2",
      "created_at": "2021-10-27T07:16:02Z"
    },
    {
      "value": "1",
      "created_at": "2021-10-27T07:16:02Z"
    }
  ],
  "success": true,
  "errors": [],
  "messages": []
}

-----------------------------------------------------: timestamp=2021-10-27T09:16:36.630+0200
2021-10-27T09:16:36.634+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-10-27T09:16:36.637+0200 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.3.0/darwin_amd64/terraform-provider-cloudflare_v3.3.0 pid=19797
2021-10-27T09:16:36.637+0200 [DEBUG] provider: plugin exited
2021-10-27T09:16:36.637+0200 [INFO]  backend/local: plan operation completed

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # cloudflare_teams_list.test_list will be updated in-place
  ~ resource "cloudflare_teams_list" "test_list" {
        id          = "002e2787-****-****-****-f88baba7a772"
      ~ items       = [
            # (1 unchanged element hidden)
            "2",
          + "3",
            "4",
            "5",
          - "3",
        ]
        name        = "Test list"
        # (3 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

### Panic output

_No response_

### Expected output

```
Plan: 0 to add, 0 to change, 0 to destroy.
```

### Actual output

```
  # cloudflare_teams_list.test_list will be updated in-place
  ~ resource "cloudflare_teams_list" "test_list" {
        id          = "002e2787-****-****-****-f88baba7a772"
      ~ items       = [
            # (1 unchanged element hidden)
            "2",
          + "3",
            "4",
            "5",
          - "3",
        ]
        name        = "Test list"
        # (3 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### Steps to reproduce

1. Create a `cloudflare_teams_list` resource containing list of items e.g. `1,2,4,5`
2. Run `terraform apply` to create resource
3. Change resource by adding item 3 to the list, while keeping the list ordered `1,2,3,4,5`
4. Run `terraform apply` to apply changes to resource
5. Run `terraform plan`

### Additional factoids

_No response_

### References

_No response_

@jacobbednarz huh when i try applying change it crashes

```
igor@Igor-Ajdiseks-MacBook-Pro test % terraform apply
╷
│ Warning: Provider development overrides are in effect
│
│ The following provider development overrides are set in the CLI configuration:
│  - cloudflare/cloudflare in /Users/igor/go/src/github.com/cloudflare/terraform-provider-cloudflare
│
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.
╵
cloudflare_teams_list.test_list: Refreshing state... [id=5deedfb6-****-****-****-f25e772a8f6a]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # cloudflare_teams_list.test_list will be updated in-place
  ~ resource "cloudflare_teams_list" "test_list" {
        id          = "5deedfb6-****-****-****-f25e772a8f6a"
      ~ items       = [
          + "3",
            # (4 unchanged elements hidden)
        ]
        name        = "Test list"
        # (3 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

cloudflare_teams_list.test_list: Modifying... [id=5deedfb6-****-****-****-f25e772a8f6a]
╷
│ Error: Plugin did not respond
│
│   with cloudflare_teams_list.test_list,
│   on teams_lists.tf line 1, in resource "cloudflare_teams_list" "test_list":
│    1: resource "cloudflare_teams_list" "test_list" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-cloudflare plugin:

panic: interface conversion: interface {} is *schema.Set, not []interface {}

goroutine 98 [running]:
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.expandInterfaceToStringList({0x18e67c0, 0xc000360360})
	/Users/igor/go/src/github.com/cloudflare/terraform-provider-cloudflare/cloudflare/utils.go:19 +0x174
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareTeamsListUpdate(0xc00039d700, {0x1911860, 0xc000188540})
	/Users/igor/go/src/github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_teams_list.go:105 +0x5e8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x1820300, {0x1a431f8, 0xc00026a580}, 0x24, {0x1911860, 0xc000188540})
	/Users/igor/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:363 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000491880, {0x1a431f8, 0xc00026a580}, 0xc0000aa4e0, 0xc00039d580, {0x1911860, 0xc000188540})
	/Users/igor/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:475 +0x6ba
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000116120, {0x1a431f8, 0xc00026a580}, 0xc00009ee10)
	/Users/igor/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000122880, {0x1a432a0, 0xc000388210}, 0xc00091c230)
	/Users/igor/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:603 +0x30e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x18e6660, 0xc000122880}, {0x1a432a0, 0xc000388210}, 0xc000736240, 0x0)
	/Users/igor/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001388c0, {0x1a518a8, 0xc0003fc000}, 0xc000894000, 0xc0004c1260, 0x1f8e5e0, 0x0)
	/Users/igor/go/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0001388c0, {0x1a518a8, 0xc0003fc000}, 0xc000894000, 0x0)
	/Users/igor/go/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/Users/igor/go/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/igor/go/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:939 +0x294
```

--- a/.changelog/1338.txt
+++ b/.changelog/1338.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/cloudflare_teams_list: ignore `items` ordering
+```

--- a/cloudflare/resource_cloudflare_teams_list.go
+++ b/cloudflare/resource_cloudflare_teams_list.go
@@ -32,7 +32,7 @@ func resourceCloudflareTeamsListCreate(d *schema.ResourceData, meta interface{})
 		Description: d.Get("description").(string),
 	}
 
-	itemValues := d.Get("items").([]interface{})
+	itemValues := d.Get("items").(*schema.Set).List()
 	for _, v := range itemValues {
 		newTeamsList.Items = append(newTeamsList.Items, cloudflare.TeamsListItem{Value: v.(string)})
 	}
@@ -102,8 +102,8 @@ func resourceCloudflareTeamsListUpdate(d *schema.ResourceData, meta interface{})
 
 	if d.HasChange("items") {
 		oldItemsIface, newItemsIface := d.GetChange("items")
-		oldItems := expandInterfaceToStringList(oldItemsIface)
-		newItems := expandInterfaceToStringList(newItemsIface)
+		oldItems := oldItemsIface.(*schema.Set).List()
+		newItems := newItemsIface.(*schema.Set).List()
 		patchTeamsList := cloudflare.PatchTeamsList{ID: d.Id()}
 		setListItemDiff(&patchTeamsList, oldItems, newItems)
 		l, err := client.PatchTeamsList(context.Background(), accountID, patchTeamsList)
@@ -153,13 +153,13 @@ func resourceCloudflareTeamsListImport(d *schema.ResourceData, meta interface{})
 	return []*schema.ResourceData{d}, nil
 }
 
-func setListItemDiff(patchList *cloudflare.PatchTeamsList, oldItems []string, newItems []string) {
+func setListItemDiff(patchList *cloudflare.PatchTeamsList, oldItems, newItems []interface{}) {
 	counts := make(map[string]int)
 	for _, val := range newItems {
-		counts[val] += 1
+		counts[val.(string)] += 1
 	}
 	for _, val := range oldItems {
-		counts[val] -= 1
+		counts[val.(string)] -= 1
 	}
 
 	for key, val := range counts {

--- a/cloudflare/resource_cloudflare_teams_list_test.go
+++ b/cloudflare/resource_cloudflare_teams_list_test.go
@@ -47,6 +47,36 @@ func TestAccCloudflareTeamsListBasic(t *testing.T) {
 	})
 }
 
+func TestAccCloudflareTeamsListReordered(t *testing.T) {
+	// Temporarily unset CLOUDFLARE_API_TOKEN if it is set as the Access
+	// service does not yet support the API tokens and it results in
+	// misleading state error messages.
+	if os.Getenv("CLOUDFLARE_API_TOKEN") != "" {
+		defer func(apiToken string) {
+			os.Setenv("CLOUDFLARE_API_TOKEN", apiToken)
+		}(os.Getenv("CLOUDFLARE_API_TOKEN"))
+		os.Setenv("CLOUDFLARE_API_TOKEN", "")
+	}
+
+	rnd := generateRandomResourceName()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccessAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareTeamsListDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareTeamsListConfigBasic(rnd, accountID),
+			},
+			{
+				Config: testAccCloudflareTeamsListConfigReorderedItems(rnd, accountID),
+			},
+		},
+	})
+}
+
 func testAccCloudflareTeamsListConfigBasic(rnd, accountID string) string {
 	return fmt.Sprintf(`
 resource "cloudflare_teams_list" "%[1]s" {
@@ -55,6 +85,18 @@ resource "cloudflare_teams_list" "%[1]s" {
 	description = "My description"
 	type        = "SERIAL"
 	items       = ["asdf-1234", "asdf-5678"]
+}
+`, rnd, accountID)
+}
+
+func testAccCloudflareTeamsListConfigReorderedItems(rnd, accountID string) string {
+	return fmt.Sprintf(`
+resource "cloudflare_teams_list" "%[1]s" {
+	account_id  = "%[2]s"
+	name        = "%[1]s"
+	description = "My description"
+	type        = "SERIAL"
+	items       = ["asdf-5678", "asdf-1234"]
 }
 `, rnd, accountID)
 }

--- a/cloudflare/schema_cloudflare_teams_list.go
+++ b/cloudflare/schema_cloudflare_teams_list.go
@@ -25,7 +25,7 @@ func resourceCloudflareTeamsListSchema() map[string]*schema.Schema {
 			Optional: true,
 		},
 		"items": {
-			Type:     schema.TypeList,
+			Type:     schema.TypeSet,
 			Optional: true,
 			Elem: &schema.Schema{
 				Type: schema.TypeString,

